### PR TITLE
RestoreFootprintOp

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

Resolves https://github.com/stellar/rs-soroban-env/issues/863.

A test was added, but expiration enforce will follow in a different PR - https://github.com/stellar/stellar-core/pull/3791. More tests should be added with that. 

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/transactions/RestoreFootprintOpFrame.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    {
+        innerResult().code(RESTORE_FOOTPRINT_MALFORMED);
+        return false;
+    }
+
+    for (auto const& lk : footprint.readWrite)
+    {
+        if (!isSorobanEntry(lk) || isTemporaryEntry(lk))
+        {
+            innerResult().code(RESTORE_FOOTPRINT_MALFORMED);
```
Comment (SirTyson) [inline src/transactions/RestoreFootprintOpFrame.cpp:123]: Should be `!isSorobanDataEntry` since we should only support `DATA_ENTRY` keys.

Comment (sisuresh) [reply src/transactions/RestoreFootprintOpFrame.cpp:123]: Good catch. Fixed that same bug in the bump op as well

File: src/testdata/ledger-close-meta-v1-protocol-20.json [hunk 2]
```diff
@@ -3,24 +3,24 @@
         "v": 1,
         "v1": {
             "ledgerHeader": {
-                "hash": "a24f02f4efbe8eb6f16236d83e78a659e97120fdc90daa6078675cfb866f2b17",
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/testdata/ledger-close-meta-v1-protocol-20.json:6]: This is due to the config settings change in xdr.

File: src/main/main.cpp [hunk 3]
```diff
@@ -263,7 +263,7 @@ main(int argc, char* const* argv)
     checkStellarCoreMajorVersionProtocolIdentity();
     rust_bridge::check_lockfile_has_expected_dep_trees(
         Config::CURRENT_LEDGER_PROTOCOL_VERSION);
-    checkXDRFileIdentity();
+    // checkXDRFileIdentity();
```
Comment (dmkozh) [inline src/main/main.cpp:266]: Please revert this

Comment (sisuresh) [reply src/main/main.cpp]: Ah good catch. Fixed

**Comment by dmkozh** [comment]
r+ f6a420c

**Comment by dmkozh** [comment]
r+ e821277

**Pull request merged**